varcode/effect_collection.py

```diff
@@ -79,13 +151,17 @@ def summary_string(self):
             # if we only printed one effect for this gene then
             # it's redundant to print it again as the highest priority effect
             if len(variant_effects) > 1:
-                # since summary effect also calls into Variant.effects,
-                # give it the same arguments
-                # (this is the downside of not having a VariantEffectCollection)
-                best = effect.top_effect()
+                best = variant_effects.top_priority_effect()
                 lines.append("  Highest Priority Effect: %s" % best)
         return "\n".join(lines)
 
+    def show(self):
+        """
+        Print effects in this collection, grouped by mutation and
+        sub-grouped by gene
+        """
+        print(self.summary_string())
```

Comment:
I switched the `__str__` method to actually just print the first 100 collection elements, so don't need these any more.